src/core/tls.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2015 Alexander Færøy <ahf@irssi.org>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301,USA
+ */
+
+#ifndef __TLS_H
+#define __TLS_H
+
+#include <openssl/ssl.h>
+
+#include <stdbool.h>
+
+typedef struct _TLS_REC TLS_REC;
+typedef struct _TLS_CERT_REC TLS_CERT_REC;
+typedef struct _TLS_CERT_ENTRY_REC TLS_CERT_ENTRY_REC;
+
+struct _TLS_REC {
+	char *protocol_version;
+	char *cipher;
+	size_t cipher_size;
+
+	char *public_key_algorithm;
+	char *public_key_fingerprint;
+	char *public_key_fingerprint_algorithm;
+	size_t public_key_size;
+
+	char *certificate_fingerprint;
+	char *certificate_fingerprint_algorithm;
+
+	char *not_after;
+	char *not_before;
+
+#ifdef SSL_get_server_tmp_key
```

Comment:
Maybe it would be better to keep these fields in the struct even if SSL_get_server_tmp_key isn't defined